### PR TITLE
Updated the YAML parsing method

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #3483

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Synchronized the process of loading and parsing YAML databases.
  * Provides more robust parse warnings/errors.
  * Added Header checks to validate database versions.
  * Adjusted root keys to a standard name of Body.
  * Adjusted inter-server storages, achievements, and attendance to follow the new format.
Thanks to @Lemongrass3110!

### CODE REVIEW

File: conf/inter_server.yml [hunk 1]
```diff
@@ -28,7 +28,11 @@
 #	  Table: "<storage_table>"  // (string) Name of table where storage is saved. The table stucture is the same as the default storage table.
 #	  Max: <max_amount>         // (int) *optional* Maximum number of items in storage. MAX_STORAGE will be used if no value is defined.
 
-Storages:
+Header:
+  Type: INTER_SERVER_DB
+  Version: 1
```
Comment (Atemo) [inline conf/inter_server.yml]: Just nothing : the indentation doesn't match, this file is using 4 spaces.

File: src/char/inter.cpp [hunk 2]
```diff
@@ -848,89 +849,75 @@ int inter_log(const char* fmt, ...)
...
+	Database db("INTER_SERVER_DB", 1);
 
-				count++;
-			}
-		}
-		ShowStatus("Done reading '" CL_WHITE "%d" CL_RESET "' storage information in '" CL_WHITE "%s" CL_RESET "'\n", count, current_file.c_str());
-	}
+	if (!db.parse(directories, parse_t(inter_server_read_db_sub)))
+		return;
 }
```
Comment (Atemo) [inline src/char/inter.cpp]: The return isn't useless here?

**Comment by Lemongrass3110** [comment]
This fixes #3483

File: src/common/database.cpp [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+bool Database::load(const std::string& path) {
+	YAML::Node rootNode;
+
+	try {
+		rootNode = YAML::LoadFile(path);
+	}
+	catch(YAML::Exception &e) {
+		ShowError("Failed to read %s database file from '" CL_WHITE "%s" CL_RESET "'.\n", this->type.c_str(), path.c_str());
+		ShowError("%s (%d:%d)\n", e.msg.c_str(), e.mark.line, e.mark.column);
```
Comment (Atemo) [inline src/common/database.cpp]: `ShowError("%s (%d:%d)\n", e.msg.c_str(), e.mark.line, e.mark.column);` this message is not clearly understandable

File: src/char/inter.cpp (hunk 2 above)
Comment (Jeybla) [reply src/char/inter.cpp]: Maybe adding a comment in all *_read_db functions then? Like
```
// Space to do checks/updates on the initially loaded database
```

**Comment by rAthenaAPI** [comment]
This pull request **fixes 1 alert** when merging 240d5b73b6ff8a6cd14fe4cbeb25d0867b3ace79 into 757a4593201195a91440b83a9fe6dafa89e22cb8 - [view on LGTM.com](https://lgtm.com/projects/g/rathena/rathena/rev/pr-c7ec4fc7be3b32e9ad97dc68bee20101d64151ee)

**fixed alerts:**

* 1 for Comparison result is always the same

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by rAthenaAPI** [comment]
This pull request **fixes 1 alert** when merging e3eea2d9bcea35445a37fe26841be5b87880d639 into 757a4593201195a91440b83a9fe6dafa89e22cb8 - [view on LGTM.com](https://lgtm.com/projects/g/rathena/rathena/rev/pr-d3b409bc49b7b44d3ff1b16d80bce0b0822fbdf4)

**fixed alerts:**

* 1 for Comparison result is always the same

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by rAthenaAPI** [comment]
This pull request **fixes 1 alert** when merging 311b14a1455ab276a159d433550f784ebeb24ab9 into 757a4593201195a91440b83a9fe6dafa89e22cb8 - [view on LGTM.com](https://lgtm.com/projects/g/rathena/rathena/rev/pr-fb9e8d23d47b336011ed73cd0829dbb882603858)

**fixed alerts:**

* 1 for Comparison result is always the same

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**